src/syntax-tree/ast.ts

```diff
@@ -2577,6 +2577,25 @@ export class ListLiteralExpression extends Expression {
         this.performTypeUpdatesOnInsertInto(insertCode);
     }
 
+    isHolePlacementValid(): boolean {
+        const emptyHolePlacements = this.getEmptyHolesWIndex();
+        return emptyHolePlacements.length === 0
+            ? true
+            : emptyHolePlacements.length === 1 && emptyHolePlacements[0][1] === this.tokens.length - 2;
```

Comment:
Only the last hole in the list can be empty.

Reply:
excatly!